internal/discord/commands.go

```diff
@@ -16,6 +19,61 @@ func mapOptions(i *discordgo.InteractionCreate) map[string]*discordgo.Applicatio
 	return optionMap
 }
 
+// parseDuration parses the string provided and returns the time.Duration equivalent
+// does not support negative durations
+func parseDuration(userInput string) (time.Duration, error) {
+	const maxDuration time.Duration = 1<<63 - 1
+	// matches any string that is a string of numbers followed by a single letter
+	r, _ := regexp.Compile(`^([\d]+)([a-zA-Z]{1})$`)
+
+	// groups[0] is the entire match, following elements are capture groups
+	groups := r.FindStringSubmatch(userInput)
+	if len(groups) < 2 {
+		err := fmt.Errorf("invalid format")
+		return 0, err
+	}
+	num, err := strconv.ParseInt(groups[1], 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	// get duration based on unit
+	var factor time.Duration
+	switch groups[2] {
+	case "s":
+		factor = time.Second
+	case "m":
+		factor = time.Minute
+	case "h":
+		factor = time.Hour
+	case "d":
+		factor = time.Hour * 24
+	case "w":
```

Comment:
is there a reason why we're getting durations based on `seconds`, `days`, `weeks`, and `years` when we only support `minutes`, `hours`, and `days`?

Reply:
Since this was a general function I just implemented what I thought might be used at some point, but I can see that weeks/years would be redundant. Seconds is useful for testing though

Reply:
yeah i think weeks/years would be overkill. let's get rid of those and keep the seconds :)